Modules/Filtering/ImageFeature/include/itkFastBilateralImageFilter.h

```diff
@@ -0,0 +1,204 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkFastBilateralImageFilter_h
+#define itkFastBilateralImageFilter_h
+
+#include "itkImageToImageFilter.h"
+#include "itkImage.h"
+#include "itkDiscreteGaussianImageFilter.h"
+#include "itkImageRegionIterator.h"
+#include "itkImageRegionConstIteratorWithIndex.h"
+#include "itkLinearInterpolateImageFunction.h"
+
+namespace itk
+{
+/**
+ * \class FastBilateralImageFilter
+ * \brief A fast approximation to the bilateral filter
+ * \ingroup FastBilateral
+ *
+ * This filter is a fast approximation to the bilateral filter.
+ * Blurring is performed on an image based on the distance of pixels in
+ * both space and intensity.
+ *
+ * The algorithm used was originally proposed by Paris and
+ * Durand [1].
+ *
+ * Instead of calculating a kernel for every pixel in
+ * an image, this filter places the values of each pixel into a higher
+ * dimensional image determined by the position and intensity of a pixel.
+ * How many bins are used is determined by the sigma values provided
+ * to the filter. Larger sigmas will result in more aggressive downsampling
+ * and less running time overall. After the data of an image
+ * has been organized into bins, a DiscreteGaussianImageFilter is applied.
+ * Finally, the output image is constructed by interpolating the
+ * values of the output pixels from the blurred higher
+ * dimensional image.
+ *
+ * This filter is great for large spatial sigmas. Numerical differences to
+ * BilateralImageFilter are negligible for most purposes.
+ *
+ * NOTE: This filter is slow for small intensity sigmas and large pixel types
+ * (e.g. short, int, or float with large intensity range).
+ *
+ *
+ * [1] Sylvain Paris and Frédo Durand,
+ *     A Fast Approximation of the Bilateral Filter using a Signal Processing
+ *     Approach,
+ *     European Conference on Computer Vision (ECCV'06)
+ *
+ * \sa BilateralImageFilter
+ * \sa GaussianOperator
+ * \sa AnisotropicDiffusionImageFilter
+ * \sa Image
+ * \sa Neighborhood
+ * \sa NeighborhoodOperator
+ *
+ * \ingroup ImageEnhancement
+ * \ingroup ImageFeatureExtraction
+ *
+ * \todo Support for color images
+ * \todo Support for vector images
+ */
+
+template <class TInputImage, class TOutputImage>
```

Comment:
ITK started using `typename` for template parameters back in 2013, especially commit 86bbb9d092d21bb309b0260c0b4082d9af966b67 by Hans (@hjmjohnson)